vector/src/main/java/im/vector/app/core/utils/ToolbarConfig.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.utils
+
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.content.res.AppCompatResources
+import com.google.android.material.appbar.MaterialToolbar
+import im.vector.app.R
+
+/**
+ * Helper class to configure toolbar.
+ * Wraps [MaterialToolbar] providing set of methods to configure it
+ */
+class ToolbarConfig(val activity: AppCompatActivity, val toolbar: MaterialToolbar) {
+    private var customBackResId: Int? = null
+
+    fun setup() = apply {
+        activity.setSupportActionBar(toolbar)
+    }
+
+    /**
+     * Delegating property for [toolbar.title]
+     * */
+    var title: CharSequence? by toolbar::title
+
+    /**
+     * Delegating property for [toolbar.subtitle]
+     * */
+    var subtitle: CharSequence? by toolbar::subtitle
+
+    /**
+     * Sets toolbar's title text
+     * */
+    fun setTitle(title: CharSequence?) = apply { toolbar.title = title }
+
+    /**
+     * Sets toolbar's title text using provided string resource
+     * */
+    fun setTitle(@StringRes titleRes: Int) = apply { toolbar.setTitle(titleRes)}
+
+    /**
+     * Sets toolbar's subtitle text
+     * */
+    fun setSubtitle(subtitle: String?) = apply { toolbar.subtitle = subtitle }
+
+    /**
+     * Sets toolbar's title text using provided string resource
+     * */
+    fun setSubtitle(@StringRes subtitleRes: Int) = apply { toolbar.subtitle = activity?.getString(subtitleRes) }
+
+    /**
+     * Enables/disables navigate back button
+     *
+     * @param isAllowed defines if back button is enabled. Default [true]
+     * @param useCross defines if cross icon should be used instead of arrow. Default [false]
+     * */
+    fun allowBack(isAllowed: Boolean = true, useCross: Boolean = false) = apply {
```

Comment:
what happens if we don't call `allowBack`? previously we were always calling `setDisplayShowHomeEnabled/setDisplayHomeAsUpEnabled`

Reply:
Oh, I forgot to mention this. Initially I did it in way, when back was enabled by default, so there will be less boilerplate. But then, while I was reworking screens, I faced a problem when you can't say if back button enabled for a specific fragment or not, by just looking to it's code, since it was controlled by activity. So I had a thought that even though it will increase boilerplate, it's still might be valuable to force to call .allowBack if you want to show back button, since in this case it's explicit and you don't need to know about any default value/hidden calls/etc. But this can easily be reverted back to having back enabled by default and calling it only for a case when you need to not show it. What do you think about it? @bmarty could you also share you thoughts?

Reply:
I have no strong opinion, as far as there is no regression :)

Reply:
I'm also easy, explicitly adding the back option sounds like a good refactoring step

I do wonder if we should only expose a subset of options to fragments which modify toolbars, only let them update the titles/options rather than determine the back behaviour and then force (or default) all activities with a toolbar (minus the top level home) to have a back/cross action, as we'll always be going a level deeper when navigating to them

Reply:
On a previous project, I had some open funs like `isModal(): Boolean`, `getTitle(): String?`, `getSubtitle(): String?`, in the parent fragment and parent activity and all the logic was managed there. The developer only had to answer the questions when implementing a new screen, it was quite handy. Here I think it's next level, as long as it's easy to use for the developer and that default behavior is sort of the expected one.

Reply:
modal screens are still in discussion, but when we will implement, we could ofcourse take there some part of the logic, like cross icon for sure. About title and subtitle - I'd rather prefer having them in xml, when it's possibe, so it will be easier to see  together with rest part of UI. We can change mind later, but for now, if i got it correct, I will leave requirement to enable back button explicitly. Let me know if I got it wrong :)